### PR TITLE
HDX-9884 Response labelling and miscellaneous fixes

### PR DESCRIPTION
**Pull request by IanHopkinson**
This PR provides standardised descriptions for query parameters and response schema, derived from the RTD documentation. It also updates query parameters accordingly.

Many descriptions are provided in `doc_snippets.py` as query parameter descriptions, they are reused for the response schema by applying a function that strips of the query preamble of "Filter response by..."

Enums are linked to their entries in RTD.

Fixes for comments on PR #135 and issues on GitHub

### CODE REVIEW

File: hdx_hapi/config/doc_snippets.py [hunk 1]
```diff
@@ -2,55 +2,85 @@
 
 CONFIG = get_config()
 
-DOC_ADMIN1_REF = 'Filter the response by the 1st subnational administrative reference number. The admin1 reference is intended as a stable identifier which will not change if, for example, admin1 name changes.'
+# Location related
+DOC_SCOPE_DISCLAIMER = f'Not all data are available for all locations. Learn more about the scope of data coverage in HDX HAPI in the <a href="{CONFIG.HAPI_READTHEDOCS_OVERVIEW_URL}">Overview and Getting Started</a> documentation.'
+DOC_LOCATION_ID = 'Filter the response by a location ID.'
```
Comment (turnerm) [inline hdx_hapi/config/doc_snippets.py]: Maybe add to the sentence: "a unique identifier for each entry in the Location table."

Can do the same for the admin IDs.

File: hdx_hapi/config/doc_snippets.py [hunk 2]
```diff
@@ -2,55 +2,85 @@
...
+DOC_LOCATION_REF = 'Filter the response by a location (typically a country) reference number.'
```
Comment (turnerm) [inline hdx_hapi/config/doc_snippets.py:8]: What do you think of: "Filter the response by a location (typically a country),  using the unique identifier (id) from the `location` table associated with each entry".

File: hdx_hapi/config/doc_snippets.py [hunk 3]
```diff
@@ -2,55 +2,85 @@
...
-DOC_HDX_PROVIDER_STUB = "Filter the response by the code of the provider (organization) of the dataset on HDX. A URL in the pattern of `https://data.humdata.org/organization/[hdx_provider_stub]` will load the provider's page on HDX."
+DOC_HDX_PROVIDER_STUB = "Filter the response by the URL-safe name of the provider (organization) of the dataset on HDX. A URL in the pattern of `https://data.humdata.org/organization/[hdx_provider_stub]` will load the provider's page on HDX."
 DOC_HDX_PROVIDER_NAME = 'Filter the response by the display name of the provider (organization) of the dataset on HDX.'
-DOC_HDX_RESOURCE_ID = 'Filter the response by the resource ID (hdx_id), which is a unique and fixed identifier of a resource on HDX. A URL in the pattern of `https://data.humdata.org/dataset/[dataset_hdx_id]/resource/[resource_hdx_id]` will load the resource page on HDX.'
+DOC_HDX_RESOURCE_ID = 'Filter the response by the resource ID (resource_hdx_id), which is a unique and fixed identifier of a resource on HDX. A URL in the pattern of `https://data.humdata.org/dataset/[dataset_hdx_id]/resource/[resource_hdx_id]` will load the resource page on HDX.'
```
Comment (turnerm) [inline hdx_hapi/config/doc_snippets.py]: Can you remove the part in brackets?

File: hdx_hapi/config/doc_snippets.py [hunk 4]
```diff
@@ -2,55 +2,85 @@
...
-DOC_LOCATION_ID = 'Filter the response by a location ID.'
-DOC_LOCATION_REF = 'Filter the response by a location (typically a country) reference number. The location reference is intended as a stable identifier which will not change if, for example, location name changes.'
-DOC_LOCATION_CODE = 'Filter the response by a location (typically a country). The location codes use the ISO-3 (ISO 3166 alpha-3) codes.'
-DOC_LOCATION_NAME = 'Filter the response by a location (typically a country). The location names are based on the "short name" from the <a href="https://unstats.un.org/unsd/methodology/m49/#fn2">UN M49 Standard</a>.'
+DOC_HDX_PROVIDER_IN_RESOURCE_STUB = "Filter the response by the URL-safe name of the provider (organization) of the dataset on HDX. A URL in the pattern of `https://data.humdata.org/organization/[dataset_hdx_provider_stub]` will load the provider's page on HDX."
+DOC_SEE_DATASET = 'See the <a href="/docs#/Metadata/get_datasets_api_v1_metadata_dataset_get" target="_blank">dataset endpoint</a> for details.'
+
+# Time periods
+DOC_UPDATE_DATE_MIN = 'Min date of update date, e.g. 2020-01-01 or 2020-01-01T00:00:00'
```
Comment (turnerm) [inline hdx_hapi/config/doc_snippets.py]: Minimum date that datasets was last updated
or 
Min of dataset update date

Same change for max

**Review by cjhendrix — APPROVED** [review]
Thanks Ian

File: hdx_hapi/config/doc_snippets.py [hunk 5]
```diff
@@ -2,55 +2,85 @@
...
-DOC_HAPI_REPLACED_DATE_MAX = 'Max date of HDX HAPI replaced date, e.g. 2020-01-01 or 2020-01-01T00:00:00'
-
-DOC_SEE_ADMIN1 = 'See the <a href="/docs#/Metadata/get_admin1_api_v1_metadata_admin1_get" target="_blank">admin1 endpoint</a> for details.'
-DOC_SEE_ADMIN2 = 'See the <a href="/docs#/Metadata/get_admin2_api_v1_metadata_admin2_get" target="_blank">admin2 endpoint</a> for details.'
-DOC_SEE_DATASET = 'See the <a href="/docs#/Metadata/get_datasets_api_v1_metadata_dataset_get" target="_blank">dataset endpoint</a> for details.'
-DOC_SEE_LOC = 'See the <a href="/docs#/Metadata/get_locations_api_v1_metadata_location_get" target="_blank">location endpoint</a> for details.'
 DOC_SEE_ORG_TYPE = 'See the <a href="/docs#/Metadata/get_org_types_api_v1_metadata_org_type_get" target="_blank">org type endpoint</a> for details.'
 
+
+DOC_AGE_RANGE = 'Filter the response by the age range. These are expressed as [start age]-[end age]. The end age is assumed to be inclusive, though that is not always explicit in the source data.'
```
Comment (turnerm) [inline hdx_hapi/config/doc_snippets.py]: What about  "These are expressed as [start age]-[end age], or [start age]+ for an age range starting at [start age] or above.".

Also good point about the inclusivity, I'll need to add that to RTD.

**Review by danmihaila — APPROVED** [review]

**Review by turnerm — APPROVED** [review]
Doc strings look good, just some small changes

File: hdx_hapi/endpoints/models/admin_level.py [hunk 6]
```diff
@@ -2,51 +2,60 @@
...
 
     model_config = ConfigDict(from_attributes=True)
 
 
 class Admin1Response(HapiBaseModel):
     id: int
     location_ref: int
-    code: str = Field(max_length=128)
-    name: str = Field(max_length=512)
-    from_cods: bool
```
Comment (danmihaila) [inline hdx_hapi/endpoints/models/admin_level.py:23]: from_cods needs to be in response

Comment (danmihaila) [reply hdx_hapi/endpoints/models/admin_level.py:23]: tests are failing, from_cods are not in response

File: hdx_hapi/config/doc_snippets.py (hunk 1 above)
Comment (IanHopkinson) [reply hdx_hapi/config/doc_snippets.py]: Fixed this one.

File: hdx_hapi/config/doc_snippets.py (hunk 3 above)
Comment (IanHopkinson) [reply hdx_hapi/config/doc_snippets.py]: Yes - done.

File: hdx_hapi/config/doc_snippets.py (hunk 4 above)
Comment (IanHopkinson) [reply hdx_hapi/config/doc_snippets.py]: Fixed this one.

File: hdx_hapi/config/doc_snippets.py (hunk 5 above)
Comment (IanHopkinson) [reply hdx_hapi/config/doc_snippets.py]: Fixed.

File: hdx_hapi/config/doc_snippets.py (hunk 2 above)
Comment (IanHopkinson) [reply hdx_hapi/config/doc_snippets.py:8]: Fixed this and changed admin1 and admin2 ID text similarly

File: hdx_hapi/endpoints/models/admin_level.py (hunk 6 above)
Comment (IanHopkinson) [reply hdx_hapi/endpoints/models/admin_level.py:23]: @danmihaila I've fixed the failing tests and also updated text following @turnerm's suggestions

**Review by ccataalin — APPROVED** [review]

**Review by alexandru-m-g — APPROVED** [review]

**Pull request merged**